### DISCUSSION TITLE
Can I get the current proxy port in the request method?

### DISCUSSION THREAD

**Discussion by wangballball** [mitmproxy/mitmproxy#5170] [Q&A] [answered]
My needs need to set different proxy for modifying different request headers for different tasks. 

My proxy class is dynamically created, and the name is "Rewrite_" plus the current proxy port. I hope that only the request initiated by the current proxy port will go through the processing in the request method of the proxy class.

How can I get the current proxy port in the request method?

Thanks!

<img width="803" alt="image" src="https://user-images.githubusercontent.com/39551720/157843253-4553fe0f-8a8f-433a-936c-b57522c8399c.png">

**Comment by Prinzhorn** [comment, marked as answer]
It's available via `ctx.options.listen_port`

**Reply by wangballball** [reply]
Thank you very much for your help !

**Reply by wangballball** [reply]
Hello, I found that I opened multiple tasks (proxies) at the same time through multi-threading, but through `ctx.options.listen_port` I can only get the port of the latest opened proxy. Is there any way to solve it?

**Reply by mhils** [reply]
Multi-threading from within Python is simply not supported.

**Reply by wangballball** [reply]
OK. Thanks a lot
